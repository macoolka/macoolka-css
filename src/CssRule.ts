export  * from 'macoolka-css-rule'
import { StandardProperties, SvgProperties, VendorLonghandProperties } from 'csstype';
import {Rule} from 'macoolka-css-rule'
/**
 * The Provide MDN CssProperties
 * @since 0.2.0
 */
export type BaseProps = StandardProperties<string> & SvgProperties<string> & VendorLonghandProperties<string>;
/**
 * Base Rule is output that is origial css
 * All Rule will final parse to the rule
 * @desczh
 * 基本规则输出是原始的css
 * 所有的规则最后解析到这里
 * @desczh
 */
export type BaseRule = Rule<{}, {}, BaseProps, {}>