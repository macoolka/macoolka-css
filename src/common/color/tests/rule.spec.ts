
import {  ruleModule } from '../index'
import { parseProp } from '../../../CssRule';
const parse = parseProp(ruleModule);

describe('parse color', () => {
    it('should parse property mkTextColor', () => {
        expect(parse({
            mkTextColor: 'main',
        })).toEqual({
            color: '#00796b'
        });
        expect(parse({
            mkTextColor: 'accent',
        })).toEqual({
            color: '#f50057'
        });
        expect(parse({
            mkTextColor: 'primary',
        })).toEqual({
            color: 'rgba(0, 0, 0, 0.87)'
        });
        expect(parse({
            mkTextColor: {
               name: 'primary',
               inverted:true,
            }
        })).toEqual({
            color: '#fff'
        });
        expect(parse({
            mkTextColor: {
               name: 'main',
               opacity:'largeX',
            }
        })).toEqual({
            color: 'rgba(0, 121, 107, 0.7)'
        });
    })
    it('should parse property mkTextColor', () => {
        expect(parse({
            mkOpacity: 'largeXX',
        })).toEqual({
            opacity:0.87
        });
        expect(parse({
            mkOpacity: 'full',
        })).toEqual({
            opacity: 1
        });
    })
     it('should parse property mkTextColors', () => {
        expect(parse({
            mkTextColor: 'yellow',
        })).toEqual({"color": "#fbc02d"});
    }) 
    it('should parse property mkColor', () => {
        expect(parse({
            mkColor: 'main',
        })).toEqual({
            "backgroundColor": "#26a69a", "color": "#fff"
        });
        expect(parse({
            mkColor: 'primary',
        })).toEqual({
            "backgroundColor": "#fff", "color": "rgba(0, 0, 0, 0.87)"
        });
        expect(parse({
            mkColor:'transparent',
        })).toEqual({
            "backgroundColor": "transparent"
        });
       
        expect(parse({
            mkColor: {
               name: 'main',
               opacity:'largeX',
            }
        })).toEqual({
            backgroundColor: 'rgba(38, 166, 154, 0.7)',
            "color": "#fff"
        });

    })
    it('should parse property mkBorderColor', () => {
        expect(parse({
            mkBorderColor: 'main',
        })).toEqual({
            "borderColor": "#00796b"
        });
        expect(parse({
            mkBorderColor: 'primary',
        })).toEqual({
            "borderColor": "rgba(0, 0, 0, 0.87)"
        });
        expect(parse({
            mkBorderColor: {
               name: 'main',
               opacity:'largeX',
            }
        })).toEqual({
            borderColor: 'rgba(0, 121, 107, 0.7)',
        });
        expect(parse({
            mkTextColor:'main',
            mkBorderColor: {
               name: 'currentColor',
               opacity:'largeX',
            }
        })).toEqual({
            color: '#00796b',
            borderColor: 'rgba(0, 121, 107, 0.7)',
        });
        expect(parse({
            mkTextColor:'primary',
            mkBorderColor: {
               name: 'currentColor',
               opacity:'largeX',
            }
        })).toEqual({
            color: 'rgba(0, 0, 0, 0.87)',
            borderColor: 'rgba(0, 0, 0, 0.61)',
        });
        expect(parse({
           
            mkBorderColor: {
               name: 'currentColor',
               opacity:'largeX',
            }
        })).toEqual({
            borderColor: 'rgba(0, 0, 0, 0.61)',
        });
    })
    it('should parse property mkColor inverted', () => {
        expect(parse({
            mkColor: {
               name: 'main',
               inverted:true
            }
        })).toEqual({
            "backgroundColor": "#00796b", "color": "#fff"
        }); 
        expect(parse({
            mkColor: {
               name: 'divide',
               inverted:true
            }
        })).toEqual({
            "backgroundColor": "#757575", "color": "#fff"
        });
        expect(parse({
            mkColor: {
                name:'divide',
                level:'dark'
            },
        })).toEqual({
            "backgroundColor": "#bdbdbd", "color": "rgba(0, 0, 0, 0.87)"
        });

    })
    it('should parse property mkColor opacity', () => {
        expect(parse({
            mkColor: {
               name: 'main',
               opacity:'large',
            }
        })).toEqual({
            "backgroundColor": "rgba(38, 166, 154, 0.54)", "color": "#fff"
        }); 
        expect(parse({
            mkColor: {
               name: 'divide',
               opacity:'mini',
            }
        })).toEqual({
            "backgroundColor": "rgba(224, 224, 224, 0.08)", "color": "rgba(0, 0, 0, 0.87)"
        });

    })
    it('should parse property mkColor using mkTextColor opacity', () => {
        expect(parse({
            mkTextColor:'main',
            mkColor: {

               opacity:'large',
            }
        })).toEqual({
            "backgroundColor": "rgba(38, 166, 154, 0.54)", "color": "#fff"
        }); 
        expect(parse({
            mkTextColor:'primary',
            mkColor: {
               opacity:'mini',
            }
        })).toEqual({
            "backgroundColor": "rgba(255, 255, 255, 0.08)", "color": "rgba(0, 0, 0, 0.87)"
        });

    })
    it('should parse property mkColor using mkTextColor opacity', () => {
        expect(parse({
            mkTextColor:'main',
            mkBgColor: {

               opacity:'large',
            }
        })).toEqual({
            "backgroundColor": "rgba(38, 166, 154, 0.54)", "color": "#00796b"
        }); 
        expect(parse({
            mkTextColor:'primary',
            mkBgColor: {
               opacity:'mini',
            }
        })).toEqual({
            "backgroundColor": "rgba(255, 255, 255, 0.08)", "color": "rgba(0, 0, 0, 0.87)"
        });

    })
    
    it('should parse property mkColor and mkTextColor', () => {
        expect(parse({
            mkColor: 'main',
            mkTextColor:'secondary',
        })).toEqual({
            "backgroundColor": "#26a69a", "color": "#fff"
        });
        expect(parse({
            mkColor: 'accent' ,color:'red'
        })).toEqual({
            "backgroundColor": "#f50057", "color": "red"
        });

    })
     it('should parse property mkColors', () => {

        expect(parse({
            mkColor: 'yellow',
        })).toEqual({"backgroundColor": "#ffee58", "color": "rgba(0, 0, 0, 0.87)"});
    }) 
    it('parse bgColor Snapshot', () => {
        expect(parse({
            mkColor: 'primary',
        })).toMatchSnapshot();
        expect(parse({
            mkColor: 'secondary',
        })).toMatchSnapshot();
        expect(parse({
            mkColor: 'main',
        })).toMatchSnapshot();
        expect(parse({
            mkColor: 'success',
        })).toMatchSnapshot();
        expect(parse({
            mkColor: 'warning',
        })).toMatchSnapshot();
        expect(parse({
            mkColor: 'alert',
        })).toMatchSnapshot();
        expect(parse({
            mkColor: 'accent',
        })).toMatchSnapshot();
        expect(parse({
            mkColor: 'divide',
        })).toMatchSnapshot();
    })
    it('parse mkBorderColor Snapshot', () => {
        expect(parse({
            mkBorderColor: 'main',
        })).toMatchSnapshot();
        expect(parse({
            mkBorderColor: 'success',
        })).toMatchSnapshot();
        expect(parse({
            mkBorderColor: 'warning',
        })).toMatchSnapshot();
        expect(parse({
            mkBorderColor: 'alert',
        })).toMatchSnapshot();
        expect(parse({
            mkBorderColor: 'accent',
        })).toMatchSnapshot();

        expect(parse({
            mkBorderColor: 'divide',
        })).toMatchSnapshot();

        expect(parse({
            mkBorderColor: 'disabled',
        })).toMatchSnapshot();    
    })
    it('parse mkTextColor Snapshot', () => {

        expect(parse({
            mkTextColor: 'main',
        })).toMatchSnapshot();
        expect(parse({
            mkTextColor: 'success',
        })).toMatchSnapshot();
        expect(parse({
            mkTextColor: 'warning',
        })).toMatchSnapshot();
        expect(parse({
            mkTextColor: 'alert',
        })).toMatchSnapshot();
        expect(parse({
            mkTextColor: 'accent',
        })).toMatchSnapshot();

        expect(parse({
            mkTextColor: 'primary',
        })).toMatchSnapshot();
        expect(parse({
            mkTextColor: 'secondary',
        })).toMatchSnapshot();
        expect(parse({
            mkTextColor: 'disabled',
        })).toMatchSnapshot();     
    })
});

