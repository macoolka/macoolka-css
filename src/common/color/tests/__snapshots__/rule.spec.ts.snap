// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse color parse bgColor Snapshot 1`] = `
Object {
  "backgroundColor": "#fff",
  "color": "rgba(0, 0, 0, 0.87)",
}
`;

exports[`parse color parse bgColor Snapshot 2`] = `
Object {
  "backgroundColor": "#fafafa",
  "color": "rgba(0, 0, 0, 0.87)",
}
`;

exports[`parse color parse bgColor Snapshot 3`] = `
Object {
  "backgroundColor": "#26a69a",
  "color": "#fff",
}
`;

exports[`parse color parse bgColor Snapshot 4`] = `
Object {
  "backgroundColor": "#66bb6a",
  "color": "rgba(0, 0, 0, 0.87)",
}
`;

exports[`parse color parse bgColor Snapshot 5`] = `
Object {
  "backgroundColor": "#ffee58",
  "color": "rgba(0, 0, 0, 0.87)",
}
`;

exports[`parse color parse bgColor Snapshot 6`] = `
Object {
  "backgroundColor": "#ef5350",
  "color": "rgba(0, 0, 0, 0.87)",
}
`;

exports[`parse color parse bgColor Snapshot 7`] = `
Object {
  "backgroundColor": "#f50057",
  "color": "#fff",
}
`;

exports[`parse color parse bgColor Snapshot 8`] = `
Object {
  "backgroundColor": "#e0e0e0",
  "color": "rgba(0, 0, 0, 0.87)",
}
`;

exports[`parse color parse mkBorderColor Snapshot 1`] = `
Object {
  "borderColor": "#00796b",
}
`;

exports[`parse color parse mkBorderColor Snapshot 2`] = `
Object {
  "borderColor": "#388e3c",
}
`;

exports[`parse color parse mkBorderColor Snapshot 3`] = `
Object {
  "borderColor": "#fbc02d",
}
`;

exports[`parse color parse mkBorderColor Snapshot 4`] = `
Object {
  "borderColor": "#d32f2f",
}
`;

exports[`parse color parse mkBorderColor Snapshot 5`] = `
Object {
  "borderColor": "#f50057",
}
`;

exports[`parse color parse mkBorderColor Snapshot 6`] = `
Object {
  "borderColor": "#e0e0e0",
}
`;

exports[`parse color parse mkBorderColor Snapshot 7`] = `
Object {
  "borderColor": "rgba(0, 0, 0, 0.38)",
}
`;

exports[`parse color parse mkTextColor Snapshot 1`] = `
Object {
  "color": "#00796b",
}
`;

exports[`parse color parse mkTextColor Snapshot 2`] = `
Object {
  "color": "#388e3c",
}
`;

exports[`parse color parse mkTextColor Snapshot 3`] = `
Object {
  "color": "#fbc02d",
}
`;

exports[`parse color parse mkTextColor Snapshot 4`] = `
Object {
  "color": "#d32f2f",
}
`;

exports[`parse color parse mkTextColor Snapshot 5`] = `
Object {
  "color": "#f50057",
}
`;

exports[`parse color parse mkTextColor Snapshot 6`] = `
Object {
  "color": "rgba(0, 0, 0, 0.87)",
}
`;

exports[`parse color parse mkTextColor Snapshot 7`] = `
Object {
  "color": "rgba(0, 0, 0, 0.54)",
}
`;

exports[`parse color parse mkTextColor Snapshot 8`] = `
Object {
  "color": "rgba(0, 0, 0, 0.38)",
}
`;
